framework/test/FileFixtureTrait.php

```diff
@@ -35,25 +35,26 @@
      * The file should return the data array that will be stored in [[data]] after inserting into the database.
      *
      * @param string $file the data file path
+     * @param bool $throwException whether to throw exception if fixture data file does not exist.
      * @return array the data to be put into the database
      * @throws InvalidConfigException if the specified data file does not exist.
      */
-    protected function loadData($file)
+    protected function loadData($file, $throwException = true)
     {
-        if ($file === false || $file === null) {
-            return [];
+        if ($file) {
```

Comment:
@leandrogehlen, @samdark How about this structure of the method?

```php
if (!$file) {
    return [];
}

if (basename($file) === $file && $this->dataDirectory !== null) {
    $file = $this->dataDirectory . '/' . $file;
}

$file = Yii::getAlias($file);
if (is_file($file)) {
    return require $file;
} elseif ($throwException) {
    throw new InvalidConfigException("Fixture data file does not exist: {$file}");
}
```

Reply:
```php
if (!$file) {
    return [];
}

if (basename($file) === $file && $this->dataDirectory !== null) {
    $file = $this->dataDirectory . '/' . $file;
}

$file = Yii::getAlias($file);
if (is_file($file)) {
    return require $file;
}

if ($throwException) {
    throw new InvalidConfigException("Fixture data file does not exist: {$file}");
}
```

Reply:
I think this not works, when file does not exists

Reply:
must be :

```php
if (!$file) {
    return [];
}

if (basename($file) === $file && $this->dataDirectory !== null) {
    $file = $this->dataDirectory . '/' . $file;
}

$file = Yii::getAlias($file);
if (is_file($file)) {
    return require $file;
} elseif ($throwException) {
    throw new InvalidConfigException("Fixture data file does not exist: {$file}");
} else {
   return [];
}

Reply:
I think these `else` statement and `return []` are unnecessary. The test should be green after the implementation of my example.

```php
if (is_file($file)) {
    return require $file;
} elseif ($throwException) {
    throw new InvalidConfigException("Fixture data file does not exist: {$file}");
} else {
   return [];
}
```